src/pip/_internal/cli/parser.py

```diff
@@ -112,15 +112,18 @@ class UpdatingDefaultsHelpFormatter(PrettyHelpFormatter):
     """
 
     def expand_default(self, option):
-        default_value = None
+        default_values = None
         if self.parser is not None:
             self.parser._update_defaults(self.parser.defaults)
-            default_value = self.parser.defaults.get(option.dest)
+            default_values = self.parser.defaults.get(option.dest)
         help_text = optparse.IndentedHelpFormatter.expand_default(self, option)
 
-        if default_value and option.metavar == 'URL':
-            help_text = help_text.replace(
-                default_value, redact_auth_from_url(default_value))
+        if default_values and option.metavar == 'URL':
+            if not isinstance(default_values, list):
+                default_values = [default_values]
+            for val in default_values:
+                help_text = help_text.replace(
+                    val, redact_auth_from_url(val))
```

Comment:
Is `default_values` guaranteed to be `Union[List[str], str]`? I’m a bit uncomfortable with `not isinstance(..., list)`; in my experience it is better to flip it into `isinsntace(..., str)` (or `six.string_types`) to accomodate possible other sequence types.

Reply:
flipped it, also added a check to ensure that `default_values` is indeed a list (after the convertion), otherwise it aborts and just returns the help text